jwst/extract_1d/extract.py

```diff
@@ -2968,12 +2966,16 @@ def do_extract1d(input_model, ref_dict, smoothing_length=None,
 
         elif isinstance(input_model, (datamodels.CubeModel,
                                       datamodels.SlitModel)):
-
             slit = None
             # Replace the default value for slitname with a more accurate
             # value, if possible.
+            # next two lines are for NRS_BRIGHTOBJ
             if getattr(input_model, "name", None) is not None:
                 slitname = input_model.name
+            if input_model.meta.exposure.type == 'NRS_FIXEDSLIT':
```

Comment:
I would have put these two lines just after `slit = None`, but this is fine and can wait until later.